homeassistant/components/vallox/__init__.py

```diff
@@ -132,6 +136,19 @@ def get_uuid(self) -> UUID | None:
             raise ValueError
         return uuid
 
+    def get_next_filter_change_date(self) -> date | None:
+        """Return the next filter change date."""
+        next_filter_change_date = calculate_next_filter_change_date(self.metric_cache)
+
+        if not isinstance(next_filter_change_date, date):
+            _LOGGER.debug(
+                "Unexpected next filter change date type: %s",
```

Comment:
We’re not expecting random types here, since we are typed in `vallox_websocket_api` as well, right?

„Error fetching next …“ might be a tad more specific in this case I‘d say.

Reply:
Yeah I’ve been there myself in the past, also don’t know why it doesn’t seem to extend to the 3rd party lib 🤷‍♂️

That’s why I was suggesting to just modify the report string slightly and that we don’t need to report the type we got.

Reply:
Aaargh, but that doesn't work either 😬 

```
pylint...................................................................Failed
- hook id: pylint
- exit code: 8

************* Module homeassistant.components.vallox
homeassistant/components/vallox/__init__.py:143:8: R1705: Unnecessary "elif" after "return" (no-else-return)
```

Why is it considered unnecessary? 🤔